docs/reference_dataset/template_repository/hello-world-root-depls/cf-apps-deployments/generic-app/template/generic-app_manifest-tpl.yml

```diff
@@ -2,3 +2,19 @@ name: generic-app
 instances: (( grab secrets.generic-app.instances ))
 path: .
 
+memory: (( grab secrets.undefined_value || undefined.fallback.value || "64M" )) # as no value are defined, spruce will use the 64M default value
+
+# By default, spruce home dir is deployment directory (ie: credentials-resource/hello-world-root-depls/cf-apps-deployments/generic-app/)
+disk_quota: (( file "spruce-file-sample-from-secrets.txt" ))
+
+# To access generated files: "../../../../generated-files"
+# To access paas-templates directory, it is possible to use one of the following:
+#   - direct reference, quite similar, like  "../../../../additional-resource/hello-world-root-depls/cf-apps-deployments/generic-app/template"
+#   - use a variable see sample with timeout and timeout_from_file $CUSTOM_SCRIPT_DIR: (( grab $CUSTOM_SCRIPT_DIR )), in this sample, CUSTOM_SCRIPT_DIR=additional-resource/hello-world-root-depls/cf-apps-deployments/generic-app/template
```

Comment:
suggesting to keep `CUSTOM_SCRIPT_DIR`  as an undocumented internal variable, and instead plan for a new feature to export new env variables such as:

```sh
SECRET_HOME=
TEMPLATE_HOME=
ROOT_DEPLOYMENT=hello-world-root-depls
DEPLOYMENT=generic-app
```

that would be used using spruce env var interpolation https://github.com/geofffranks/spruce/blob/master/doc/environment-variables-and-defaults.md#can-i-use-the-values-of-environment-variables-in-an-operation 

```
path_to_my_file: ((concat $TEMPLATE_HOME "/" $ROOT_DEPLOYMENT "/cf-apps-deployments/"  $DEPLOYMENT "/template" ))
```

and in shell script in 
```sh
${TEMPLATE_HOME}/{ROOT_DEPLOYMENT}/cf-apps-deployments/${DEPLOYMENT}/template
${SECRET_HOME}/{ROOT_DEPLOYMENT}/cf-apps-deployments/${DEPLOYMENT}/myfile.txt
```

Reply:
handled in https://github.com/orange-cloudfoundry/cf-ops-automation/pull/258#discussion_r286430932